### DISCUSSION TITLE
useCombobox with multiple item types

### DISCUSSION THREAD

**Discussion by Rangoons** [adobe/react-spectrum#4284] [Q&A]
Hello!
We are trying to use one combobox to render two different types of data that are returned from two different search indices. Is this possible with `useCombobox` et al?

I've gotten together to return the data from `load` in `useAsyncList` as `[...dataSet1, ...dataSet2]` and it works fine if I just render the stringified item. ie: 
```
 {(item) => (
          <Item key={item.name}>
            {JSON.stringify(item)}
          </Item>
        )}
```

however, once I start to actually determine the type of item it is & render that way, I keep running into the error `Error: No key found for item` it seems to only happen if mixed items are rendered. Is it not possible to render multiple data sets in one list or am I going about this incorrectly

**Comment by LFDanLu** [comment]
Mind sharing a codesandbox/code snippet? Does each item in your data list from `useAsyncList` have a `name`?

**Reply by Rangoons** [reply]
Yes, they both have a name property. Logging the items also did not show any items with a key missing/undefined

**Reply by Rangoons** [reply]
I can't get the sandbox to error so maybe its just back to the drawing board for me but here is a rough implementation of what I'm doing

https://codesandbox.io/s/suspicious-spence-npu9c4?file=/src/App.tsx

**Reply by Rangoons** [reply]
Update - I believe it was just a case of some of the async object's having the property of `name` but the value being `undefined`. I appreciate the help 🤝
